packages/@tinacms/datalayer/src/backend/index.ts

```diff
@@ -59,29 +93,44 @@ function MakeNodeApiHandler({
   isAuthorized,
   extraRoutes,
   databaseClient,
-}: BackendAuthProvider & { databaseClient: DatabaseClient }) {
-  const handler: NodeApiHandler = async (...params) => {
-    const [req, res] = params
+  opts,
+}: BackendAuthProvider & {
+  databaseClient: DatabaseClient
+  opts: Required<TinaBackendOptions['options']>
+}) {
+  const tinaBackendHandler: NodeApiHandler = async (req, res) => {
     // remove leading slash
-    const url = req?.url?.startsWith('/') ? req.url.slice(1) : req.url
-    const paths = url?.split('/')
-    const routes = paths?.filter((p) => p !== 'api' && p !== 'tina')
+    const path = req.url?.startsWith('/') ? req.url.slice(1) : req.url
+
+    // The domain is not important here, we just need to parse the pathName
+    const url = new URL(path, `http://${req.headers?.host || 'localhost'}`)
+
+    // Remove the basePath from the url
+    const routes = url.pathname?.replace(opts.basePath, '')?.split('/')
+
     if (typeof routes === 'string') {
       throw new Error('Please name your next api route [...routes] not [route]')
     }
-    if (!routes?.length) {
+    if (!routes?.length || routes.length === 0) {
```

Comment:
I think this is redundant
```suggestion
    if (!routes?.length) {
```